### PR TITLE
feature: extend cri apis for get envs

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did

The CRI query container information interface adds the output of the environment variables, and returns the envs  of the container in the `ContainerStatus` interface

### Ⅱ. Does this pull request fix one issue?

None.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
Done.

### Ⅳ. Describe how to verify it
1. Synchronize changes to the proto file of cri-tools.
2. Use command `crictl inspect` to view the container's environment variables.

### Ⅴ. Special notes for reviews
None.

### CODE REVIEW

File: cri/v1alpha2/cri.go [hunk 1]
```diff
@@ -774,6 +774,7 @@ func (c *CriManager) ContainerStatus(ctx context.Context, r *runtime.ContainerSt
 		Volumes:     parseVolumesFromPouch(container.Config.Volumes),
 		Resources:   parseResourcesFromPouch(resources, diskQuota),
 		QuotaId:     container.Config.QuotaID,
+		Envs:        parseEnvsFromPouch(container.Config.Env),
```
Comment (allencloud) [inline cri/v1alpha2/cri.go]: I do not think we should add `Pouch` in the function naming. Please change a proper one. @Starnop

File: cri/v1alpha2/cri_utils.go [hunk 2]
```diff
@@ -1087,6 +1087,22 @@ func parseVolumesFromPouch(containerVolumes map[string]interface{}) map[string]*
 	return volumes
 }
 
+// parseEnvsFromPouch parse Envs from []string to []*runtime.KeyValue
+func parseEnvsFromPouch(pouchEnvs []string) (criEnvs []*runtime.KeyValue) {
+	for _, env := range pouchEnvs {
+		runtimeEnv := &runtime.KeyValue{}
+		if strings.Count(env, "=") == 1 {
+			envItem := strings.Split(env, "=")
```
Comment (allencloud) [inline cri/v1alpha2/cri_utils.go]: What will happen if `env` is equal to `a=b=c`?

File: cri/v1alpha2/cri_utils.go [hunk 3]
```diff
@@ -1087,6 +1087,22 @@ func parseVolumesFromPouch(containerVolumes map[string]interface{}) map[string]*
 	return volumes
 }
 
+// parseEnvsFromPouch parse Envs from []string to []*runtime.KeyValue
+func parseEnvsFromPouch(pouchEnvs []string) (criEnvs []*runtime.KeyValue) {
```
Comment (allencloud) [inline cri/v1alpha2/cri_utils.go]: you are not parsing, you just did the translation thing.

File: cri/v1alpha2/cri.go (hunk 1 above)
Comment (YaoZengzeng) [reply cri/v1alpha2/cri.go]: @Starnop Maybe we should rename the functions like `parseEnvsFromPouch`, `parseResourcesFromPouch` in another PR, they all do the same thing :)

@allencloud WDYT?

File: cri/v1alpha2/cri_utils.go (hunk 2 above)
Comment (YaoZengzeng) [reply cri/v1alpha2/cri_utils.go]: @allencloud Actually the envs here is returned from pouchd and they should have been validated when input to pouchd.

So maybe we don't need to check again here.

File: cri/v1alpha2/cri.go (hunk 1 above)
Comment (Starnop) [reply cri/v1alpha2/cri.go]: :+1: I will do that in the next pull request.

File: cri/v1alpha2/cri_utils.go (hunk 2 above)
Comment (Starnop) [reply cri/v1alpha2/cri_utils.go]: Yeah, in fact, the case 'a=b=c' will not be passed by pouchd. 
Even if that happens, the upper level can also deal with it by key.

Comment (allencloud) [reply cri/v1alpha2/cri_utils.go]: But you are using `envItem[1]` to take `b` and ignoring `=c`, right? @Starnop

Comment (fuweid) [reply cri/v1alpha2/cri_utils.go]: `a=b=c` is valid because `b=c` is value. 

```
# vagrant @ ubuntu-xenial in ~ [20:36:09]
$ A=1=2 echo ${A}
1=2
```

Comment (allencloud) [reply cri/v1alpha2/cri_utils.go]: I sponsor @fuweid 's point of view. Your split way has potential risk. Do you agree? @Starnop

File: cri/v1alpha2/cri.go (hunk 1 above)
Comment (allencloud) [reply cri/v1alpha2/cri.go]: I still insist that we could finish in the pull request. How about `convertEnvsForCRI` or `convertContainerEnvForCRI`?

**Comment by Starnop** [comment]
@fuweid Done. :)

**Comment by allencloud** [comment]
LGTM

**Pull request merged**